add lifecycle output to update stack after rebase (#360)
buildpacks/spec: platform.md

    - The value of `io.buildpacks.lifecycle.metadata` SHALL be modified as follows
      - `run-image.reference` SHALL uniquely identify `<run-image>`
      - `run-image.top-layer` SHALL be set to the uncompressed digest of the top layer in `<run-image>`
    - The value of `io.buildpacks.stack.*` labels SHALL be modified to that of the new `run-image`

> - **If** the provided `<run-image>` is not found in `stack.run-image.image` or `stack.run-image.mirrors`:

Reviewer: ```suggestion
- **If** the provided `<run-image>` is not found in `run-image.image` or `run-image.mirrors`:
```

>       - `stack.run-image.image` SHALL be the provided `<run-image>`
>       - `stack.run-image.mirrors` SHALL be omitted

Reviewer: ```suggestion
      - `run-image.image` SHALL be the provided `<run-image>`
      - `run-image.mirrors` SHALL be omitted
```
(On newer platforms, this key is moved up one level)

      - `run-image.reference` SHALL uniquely identify `<run-image>`
      - `run-image.top-layer` SHALL be set to the uncompressed digest of the top layer in `<run-image>`
    - The value of `io.buildpacks.stack.*` labels SHALL be modified to that of the new `run-image`

>       - `stack.run-image.image` SHALL uniquely identify `<run-image>`
>       - `stack.run-image.mirrors` SHALL be EMPTY

Reviewer: ```suggestion
- **If** the provided `<run-image>` is not found in `stack.run-image.image` or `stack.run-image.mirrors`:
      - `stack.run-image.image` SHALL be the provided `<run-image>`
      - `stack.run-image.mirrors` SHALL be empty
```

>       - `stack.run-image.mirrors` SHALL be updated to include the new image mirrors

Reviewer: How will the run image mirrors be identified?

Reviewer: (During build they come from stack.toml - soon to be run.toml in future platform APIs - but we don't have this file during rebase.)

Author: Changes has been made @natalieparellano

      - `run-image.reference` SHALL uniquely identify `<run-image>`
      - `run-image.top-layer` SHALL be set to the uncompressed digest of the top layer in `<run-image>`
    - The value of `io.buildpacks.stack.*` labels SHALL be modified to that of the new `run-image`

> - **If** the provided `<run-image>` is not found in `stack.run-image.image` or `stack.run-image.mirrors`:

Reviewer 2: Sorry this might be a dumb question, but does this mean every new run image released needs to have some sort of mapping/record/reference indicated in the stack?

Reviewer: I think it depends - if you want `stack.run-image.mirrors` to have data, then yes the mapping must exist. Worth noting: future versions of the spec do away with stack.toml in favor of run.toml, which now contains a list of run images (and associated mirrors for each). More information could be found here: https://github.com/buildpacks/spec/pull/335/files#diff-e603760990971da3f77be4bb8d77c3405098f006814fd8c054d2d15f395b8330R934-R939 and here: https://github.com/buildpacks/spec/pull/335/files#diff-e603760990971da3f77be4bb8d77c3405098f006814fd8c054d2d15f395b8330R1238

@hernandanielg I'm realizing that this PR should probably be pointed at https://github.com/buildpacks/spec/pull/335 due to the aforementioned changes. Are you able to rebase? There will likely be a merge conflict.

Reviewer 2: Thank you for the clarification!
